### ISSUE TITLE
EtherCalc: Look N Feel (Theming of GUI interface)

### ISSUE THREAD

**Issue by pablo3p** [audreyt/ethercalc#174]
Hi,
I was wondering if there was a way to change the Look N Feel(as they call it in the Java world) of the GUI interface.
There is a commercial javascript based spreadsheet by Wijmo. Its very expensive!

BUT, they have a demo on their site:
http://wijmo.com/demo/spreadjs/samples/ExplorerSample/?widget=Selection&sample=Drag%20Fill

BUT I like the "look and feel" of their spreadsheet. 
It is possible to change the theme of EtherCalc?

Regards,

P

**Comment by audreyt** [comment]
Yes you can change most aspects of SocialCalc with CSS. Ideas, mockups and pull requests welcome!

**Comment by pablo3p** [comment]
Hi there,
I edited the file called SocialCalc.css, and did not see any changes.
Can you direct me to which CSS file controls the GUI element colors.

P

**Comment by pablo3p** [comment]
Hi again,
I have edited other CSS files i found in the installation directory, but nothing happens to the app itself.
When on runs the app, is it somehow precompiled or something, hence would not reference the CSS files I am modifiying.
I was able to virtually make changes to the source code which seems to have all the CSS injected inline, using firefoxes firebug.
SO i am starting to feel that one needs to understand the architecture of this app.

SO looking for input on how one can change the look-N-feel of this spreadsheet GUI.
Hope you can help me.

Regards,

P

**Comment by sank64** [comment]
I just released a new theme for ethercalc, it looks like this:
 
![skjermbilde 2015-06-03 kl 09 26 17](https://cloud.githubusercontent.com/assets/9414077/7954749/a7042774-09d2-11e5-9d42-2b1d48036856.png)

And it is released under MIT Licence.
Last version is here: https://github.com/sank64/ethercalc_theme_green/tree/master/150601

-Kévin

**Comment by sank64** [comment]
You can look at the files, compare it with the original ones (ending by "-s" in the repo) to learn what has been changed. So you can make your own. If you want the icons in an other color, I ve released them as well, in an other repo on github.
Enjoy !

**Comment by ocdtrekkie** [comment]
I'd be a little concerned emulating the Excel 2013 icon so closely and also using Excel's green color.

**Comment by audreyt** [comment]
The color looks fine to me, though the upper-leftmost icon is indeed a bit suggestive. :-)

EtherCalc's current logo is rather subpar (I can say this, as I made that logo years ago...). Perhaps we can come up with something in that spirit, as to be a bit more distinctive compared with Excel's logo?

![fv5-byiu](https://cloud.githubusercontent.com/assets/20723/7955499/5b571a78-0a0b-11e5-9bc4-45545bf99741.png)

**Comment by sank64** [comment]
Side note: The version released on github is without any logo.

**Comment by pablo3p** [comment]
Hi,
I like the Look-N-Feel you created with the Green Color!
It looks great!

But I hope you can make a few themes, so people can choose the one they like the most.

I also suggest creating a light-Blue color that is similar to the commercial App SpreadJS, see link here:
http://wijmo.com/demo/spreadjs/samples/ExplorerSample/

You have to click on one of the many squares to see a live demo of their javascript spreadsheet.

Thanks for the update!!!

P

Date: Wed, 3 Jun 2015 01:14:35 -0700

**Comment by sank64** [comment]
The biggest gap from using Ethercalc remains in learning how to use it. The most it differs from its well established competitors the less it will be used broadly. Bringing Ethercalc to everyone is a ideal for me. And despite my wish to give Ethercalc its own look'n feel, given the time I have to contribute, and how strong impact a familiar color theme can have on new users, I decided to maintain first of all the green theme. 
I would like also to stress that these themes are more than a color choice, I spent much more time on improving readability, hiding redondant icons, testing mobile devices portability.

As Ethercalc is becoming easier to personalize since more and more css properties are now contained in the language preferences it will be less time consuming for me to maintain an alternative theme like a blue theme close to SpreadJS. 

Regarding the icon, I liked your ideas, and have a draft already. Just need some time to make a mockup. Next week maybe.

-K

**Comment by codecowboy** [comment]
@sank64 Do you have a copy of your theme somewhere? I much prefer yours to the default.
